astropy/io/ascii/cds.py

```diff
@@ -119,6 +119,10 @@ def get_cols(self, lines):
                 unit = match.group('units')
                 if unit == '---':
                     col.unit = None  # "---" is the marker for no unit in CDS table
+                elif unit[0] == '[' and unit[-1] == ']':
+                    temp = re.compile(r'\[(?P<un>.*?)\]')
```

Comment:
Maybe the compile can be done once at the module level or is that premature optimization?

Reply:
Yes, I think I should do the compile at the module level because each time it sees the functional unit it will recompile.